include/type_safe/detail/aligned_union.hpp

```diff
@@ -41,11 +42,11 @@ namespace detail
         static constexpr auto size_value      = detail::max(sizeof(Types)...);
         static constexpr auto alignment_value = detail::max(alignof(Types)...);
 
-        using type = typename std::aligned_storage<size_value, alignment_value>::type;
+        alignas(alignment_value) unsigned char storage_[size_value];
```

Comment:
This is not what I meant.

Instead of using `aligned_union_t<Types..> storage` in variant or wherever, you should use `alignas(aligned_union<Types...>::alignment_value) unsigned char storage[aligned_union<Types...>::size_value]`.

(This probably also gets rid of the GCC error.)

Reply:
I think the issues here is that GCC 4.8 does not think aligned_union<Types...>::alignment_value is an integer constant. I installed an older Ununtu and GCC 4.8 and played around with it but always get an error
`error: requested alignment is not an integer constant`
If I change the alignement to be an actual integer constant like 64 it compiles fine

Reply:
Then just drop GCC 4.8. It's been ages.